R/react.R

```diff
@@ -161,18 +161,82 @@ triggerEvent <- function(inputId) {
   )
 }
 
+#' @title usage for function setInput
+#'
+#' @description  Prints a description of valid parameters for the `setInput`
+#' function
+setInputUsage <- function() {
+  message("Usage of `setInput()`: see ?shiny.react::setInput for more information")
+  message()
+  message("  setInput(<inputId string>) :: equivalent as setInput(inputId, 1)")
+  message("  setInput(<inputId string>, <integer with an index>) :: index in R (starting at 1)")
+  message("  setInput(<inputId string>, <accessor string>)")
+}
+
 #' Set input
 #'
 #' Creates a handler which can be used for `onChange` and similar props of 'React' components
 #' to set the value of a 'Shiny' input to one of the arguments passed to the handler.
 #'
 #' @param inputId 'Shiny' input ID to set the value on.
-#' @param argIdx Index of the argument to use as value.
+#' @param argIdx Index (numeric) or accessor (string) of the argument to use as value.
 #' @return A `ReactData` object which can be passed as a prop to 'React' components.
 #'
 #' @export
-setInput <- function(inputId, argIdx = 1) {
-  ReactData(
-    type = "input", id = inputId, argIdx = argIdx - 1
-  )
-}
+#' @examples
+#' setInput("some_id")
+methods::setGeneric(
+  "setInput",
+  function(inputId, argIdx = 1) {
+    setInputUsage()
+    stop("Arguments not supported")
+  }
+)
+
+#' @describeIn setInput Uses as index `argIdx = 1`
+#' @export
+#' @examples
+#' setInput("some_id", 1)
+methods::setMethod(
+  "setInput",
+  signature(inputId = "character", argIdx = "missing"),
+  function(inputId) {
+    setInput(inputId, 1)
+  }
+)
+
+#' @describeIn setInput Gets the value from index in argIdx
+#' @export
+#' @examples
+#' setInput("some_id", 2)
+methods::setMethod(
+  "setInput",
+  signature(inputId = "character", argIdx = "numeric"),
+  function(inputId, argIdx = 1) {
```

Comment:
Perhaps we shall change to 0-based indexing? It might be confusing that `setInput(id, 1)` and `setInput(id, "[0]")` are equivalent and `setInput()` can be seen as a shorthand for `JS()`, so 0-based indexing might be more natural actually. WDYT?